src/nodes.ts

```diff
@@ -283,16 +283,19 @@ export class Conditional extends Node {
 }
 
 export class Program extends Node {
+  context: ParseContext;
+
   constructor(
     line: number,
     column: number,
     start: number,
     end: number,
     raw: string,
     readonly body: Block | null,
-    readonly context: ParseContext
+    context: ParseContext
   ) {
     super('Program', line, column, start, end, raw);
+    this.context = context;
```

Comment:
Yeah, I needed to make context mutable since I modify it in the test runner. I think you need the explicit assignment in the constructor in that case, although maybe there's a shorthand I'm missing.